### ISSUE TITLE
"Error: Failed to decode current backend config" when upgrading from v0.12.24 -> v0.13.2

### ISSUE THREAD

**Issue by richstokes** [hashicorp/terraform#26124]
### Terraform Version
`Terraform v0.12.24` upgrading to `Terraform v0.13.2`

Running `terraform13 0.13upgrade` before the init everything seems fine. 

### Debug Output
```
terraform13 init
Initializing modules...

Initializing the backend...

Error: Failed to decode current backend config

The backend configuration created by the most recent run of "terraform init"
could not be decoded: unsupported attribute "lock_table". The configuration
may have been initialized by an earlier version that used an incompatible
configuration structure. Run "terraform init -reconfigure" to force
re-initialization of the backend.
```

Note: None of our `.tf` files contain any kind of reference to `lock_table` - nor do our state files.  

Doesn't seem to matter which AWS provider version I use. Tried with both `2.70.0` and `3.5.0`  

Our backend is in an S3 bucket.  

### Expected Behavior
`terraform init` should succeed OK

### Actual Behavior
See above

### Steps to Reproduce
1. `terraform init`

### Our terraform.tf

```
terraform {
  required_version = "0.13.2"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "3.5.0"
    }
  }

  backend "s3" {
    bucket   = "our-s3-bucket"
    key      = "dev"
    region   = "us-east-1"
    encrypt  = true
  }
}

provider "aws" {
  version = "3.5.0"
  region  = "us-east-1"

}

```

**Comment by richstokes** [comment]
Did an `rm -rf .terraform` to nuke the state on the local disk, re-ran the `terraform init` and it seems a lot happier now. 

Maybe some sort of race condition that led it to look at some broken version of the local disk state instead of querying the S3 backend?

**Comment by alisdair** [comment]
This looks like a very out of date local backend state. [The `lock_table` argument was deprecated almost three years ago](https://github.com/hashicorp/terraform/pull/14949) and then recently [removed altogether with an S3 backend update](https://github.com/hashicorp/terraform/pull/25134). This shipped in 0.13.

I'm closing as I believe this is not a bug, and as Terraform suggested, running `terraform init -reconfigure` should have resolved this issue. Removing the local `.terraform` cache directory is also an effective workaround. Glad you were able to get it working!
